src/test/java/Java_interview_question/Java_interview_question.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Java Interview Questions
+description: Java Interview Questions that can be helpfull & handy while revising java Concepts.
+created: 2022-10-14
+updated: 2022-10-14
+---
+
+## 1. What is Java ?
+ A high-level, object-oriented, robust, secure programming language, platform-independent, high performance, Multithreaded, and portable programming language. Java was developed by James Gosling in June 1991.
+
+## 2. Features of Java ?
+*Secured*
+*Robust*
+*Portable*
+*Object-Oriented*
+*Simple*
+*Platform independent*
+*Dynamic*
+*Multithreaded*
+
+## 3. Basic Concepts of OOPS ?
+*Polymorphism* - Polymorphism refers to many forms, or it is a process that performs a single action in different ways.
+*Abstraction*  -  process which displays only the information needed and hides the unnecessary information. We can say that the main purpose of abstraction is data hiding. 
+*Inheritence*  - Inheritance is a method in which one object acquires/inherits another object’s properties, and inheritance also supports hierarchical classification.
+*Encapsulation*
+*Object*      - An entity that has state and behavior is known as an object.
+*Java Class*  - A Java class file is a file containing Java bytecode that can be executed on the Java Virtual Machine.1
+
+## 4. What is an Object ?
+Object is the real-time entity having some state and behavior.Object is an instance of the class having the instance variables as the state of the object and the methods as the behavior of the object. The object of a class can be created by using the new keyword.
+
+## 5.Difference between C++ & Java ? 
+| C++                                              | Java                                                                                           |
+| -------------                                    | -------------                                                                                  |
+| 1.C++ is platform-dependent.                     | 1 .Java is platform-independent.                                                               |
+| 2.C++ supports multiple inheritance.             | 2.Java doesn't support multiple inheritance through class. Can be achieved by using Interface. |
+| 3.C++ supports operator overloading.             | 3.Java doesn't support operator overloading.                                                   |
+
+## 6.How many types of constructors are used in Java? 
+ **Default Constructor**: *default constructor is the one which does not accept any value. The default constructor is mainly used to initialize the instance variable with the default values.* 
+
```

Comment:
### **_Example, For Better Understanding..._**
_Code:-_
//java default constuctor//
```

> public class DefaultConstructor {
>     int id;
>     String name;
>     void display()//Default constructor//
>     {
>         System.out.println(id+" "+ "Teuton" +" "+name);
>     }
>     public static void main(String[] args) {
>         DefaultConstructor d1 = new DefaultConstructor();
>         DefaultConstructor d2 = new DefaultConstructor();
>         d1.display();
>         d2.display();
>     }
> }
> 

```